### PR TITLE
Const instead of let and other code optimizations

### PR DESCRIPTION
**Pull request by programmer4web**
fixes https://github.com/Neovici/cosmoz-grouped-list/issues/14
fixes codeclimate duplication issues.

### CODE REVIEW

**Comment by programmer4web** [comment]
@nomego I have applied the const instead of let where possible and optimized group find. This is ready for Review.

File: cosmoz-grouped-list.js [hunk 1]
```diff
@@ -397,16 +389,9 @@
...
-			this.data.some(group => {
-				const found = group.items && group.items.indexOf(item) > -1;
-				if (found) {
-					foundGroup = group;
-				}
-				return found;
-			});
-
-			return foundGroup;
+			return this.data.find(group => {
```
Comment (nomego) [inline cosmoz-grouped-list.js]: `{ return ` .. ` ; }` not needed

File: cosmoz-grouped-list.js [hunk 2]
```diff
@@ -480,10 +463,10 @@
 		},
 
 		deselectItem(item) {
-			let selectedIndex = this.selectedItems.indexOf(item),
+			const selectedIndex = this.selectedItems.indexOf(item),
 				model = this._getModelFromItem(item),
-				group = this.getItemGroup(item),
-				groupState,
+				group = this.getItemGroup(item);
+			let	groupState,
```
Comment (nomego) [inline cosmoz-grouped-list.js]: This should be declared as close to usage/as late as possible

File: cosmoz-grouped-list.js [hunk 3]
```diff
@@ -565,21 +552,16 @@
...
 			if (this._groupsMap) {
 
-				this.data.forEach(function (group) {
-					let groupState = this._groupsMap.get(group);
+				this.data.forEach(group => {
+					const groupState = this._groupsMap.get(group);
 					if (groupState) {
 						groupState.selected = false;
 					}
 				}, this);
```
Comment (nomego) [inline cosmoz-grouped-list.js]: `this` not needed with arrow function

File: cosmoz-grouped-list.js [hunk 4]
```diff
@@ -594,8 +576,8 @@
 		},
 
 		toggleCollapse(item) {
-			let model = this._getModelFromItem(item),
-				itemState = this._itemsMap.get(item);
+			const model = this._getModelFromItem(item);
+			let itemState = this._itemsMap.get(item);
```
Comment (nomego) [inline cosmoz-grouped-list.js]: Declare/set as close to usage as possible.
In this case we can avoid fetching from _itemsMap if we don't have a model

**Comment by programmer4web** [comment]
@nomego I have applied the requested fixes.

File: cosmoz-grouped-list.js [hunk 5]
```diff
@@ -255,23 +248,22 @@
...
 		},
 
 		selectorItemChanged(selector, item) {
-			let selectorId = selector.selectorId,
-				isGroup = this.isGroup(item),
-				newTemplate,
-				element,
+			const selectorId = selector.selectorId,
+				isGroup = this.isGroup(item);
+			let element,
```
Comment (nomego) [inline cosmoz-grouped-list.js]: this should be declared where you set element?

**Pull request merged**